frontend-react/src/pages/submissions/SubmissionDetails.tsx

```diff
@@ -0,0 +1,182 @@
+
+    /* Create strings from parsed values */
+    const time = `${hours}:${minutes} ${meridian.toUpperCase()}`
+    const date = `${dateTimeISO.getDate()} ${monthNames[dateTimeISO.getMonth()]} ${dateTimeISO.getFullYear()}`
+
+    return {
+        dateString: date,
+        timeString: time
+    }
+}
+
+/* 
+    A component displaying a soft gray title and content in
+    standard black text. 
+
+    @param item - the title of a property; e.g. Report ID
+    @param content - the content of a property; e.g. 000000-0000-0000-000000
+*/
+export function DetailItem({ item, content, subItem }: DetailItemProps) {
+    return (
+        <div style={{
+            'display': 'flex',
+            'flexDirection': 'column',
+            'margin': subItem ? '16px 32px' : '8px 0px'
+        }}>
+            <span>{item}</span>
+            <span>{content}</span>
+        </div>
+    );
+}
+
+/* 
+    A component displaying information about a single destination
+    returned from the history/submissions details API
+
+    @param destinationObj - a single object from the destinations array
+    in the history/submissions details API
+*/
+export function DestinationItem({ destinationObj }: DestinationItemProps) {
+    const submissionDate = generateSubmissionDate(destinationObj.sending_at) || "unsent"
+    return (
+        <div style={{
+            'display': 'flex',
+            'flexDirection': 'column',
+        }}>
+            <h2>{destinationObj.organization}</h2>
+            <DetailItem
+                item={'Transmission Date'}
+                content={submissionDate.dateString}
+                subItem
+            />
+            <DetailItem
+                item={'Transmission Time'}
+                content={submissionDate.timeString}
+                subItem
+            />
+            <DetailItem
+                item={'Records'}
+                content={destinationObj.itemCount}
+                subItem
+            />
+        </div>
+    );
+}
+
+/* 
+    The page component showcasing details about a submission to the
+    sender
+
+    @param actionId - the id tracking the ActionLog object containing
+    the information to display. Used to call the API.
+*/
+function SubmissionDetails() {
+    const organization = getStoredOrg();
+    const { actionId } = useParams<SubmissionDetailsProps>()
+    const actionDetails: ActionDetailsResource = useResource(
+        ActionDetailsResource.detail(),
+        { actionId, organization }
+    );
+    const submissionDate = generateSubmissionDate(actionDetails.submittedAt)
+
+    if (!actionDetails) {
+        return (
+            <ErrorPage type="page" />
+        )
+    } else {
+        return (
+            <div
+                className="grid-container margin-bottom-10"
+                data-testid="container">
+                <div className="grid-col-12">
+                    {/* TODO: why does the spinner take the whole dang page?! */}
+                    <NetworkErrorBoundary
```

Comment:
I think the `<NetworkErrorBoundry>` being in the same component as `useResource()` doesn't work as one might expect.

This is also why spinners might be taking up the whole page?

Maybe we need to nest the component down one layer?
```js
<NetworkErrorBoundry...>
  <Suspense...>
    <SubmissionDetails>
  </Suspense...> 
</NetworkErrorBoundry>
```
So maybe some kind outer component that contains the `<SubmissionDetails>` needs to do the wrapping?

We can chat about this, that might be easier.

Reply:
I got it figured out! Oddly enough I'm pretty sure I did the same thing to the Facility's table or Report Details page at one point after much digging in the `rest-hooks` docs